File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> ans = new ArrayList<>();
        bfs(root, ans, 0);
        return ans;
    }

    private void bfs(TreeNode currentNode, List<List<Integer>> ans, int currentLevel){
        if(currentNode == null)
            return;
        if(ans.size() < currentLevel + 1){
            for(int i=ans.size(); i < currentLevel + 1; i++){
                ans.add(new ArrayList<>());
            }
        }
        System.out.println("currentLevel: "+currentLevel);
        System.out.println("ansSize: "+ans.size());
        ans.get(currentLevel).add(currentNode.val);
        if(currentNode.left != null)
            bfs(currentNode.left, ans, currentLevel + 1);
        if(currentNode.right != null)    
            bfs(currentNode.right, ans, currentLevel + 1);     
    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        que = deque()
        ans = []
        que.append((0, root))
        previousLevel = 0
        ans.append(root.val)
        while que:
            (level, node) = que.popleft()
            if previousLevel != level:
                ans.append(node.val)
            if node.right:    
                que.append((level + 1, node.right))
            if node.left:    
                que.append((level + 1, node.left))
            previousLevel = level
        return ans    

        